tests/foreman/longrun/test_oscap.py

```diff
@@ -0,0 +1,222 @@
+        7. Create an activation-key
+        8. Add product to activation-key
+        """
+        cls.ak_name = gen_string('alpha')
+        org = entities.Organization().create()
+        cls.org_name = org.name
+        # The below repos is used for cv's.
+        repos = []
+        repo_values = [{'repo': REPOS['rhst6']['name'],
+                        'reposet': REPOSET['rhst6'],
+                        },
+                       {'repo': REPOS['rhst7']['name'],
+                        'reposet': REPOSET['rhst7'],
+                        }]
+        # step 1.2: Create new lifecycle environments
+        env = entities.LifecycleEnvironment(
+            organization=org
+        ).create()
+        cls.env_name = env.name
+        # step 2: Upload manifest
+        with open(manifests.clone(), 'rb') as manifest:
+            upload_manifest(org.id, manifest)
+        for value in repo_values:
+            # step 3.1: Enable RH repo and fetch repository_id
+            repository = entities.Repository(id=enable_rhrepo_and_fetchid(
+                basearch='x86_64',
+                org_id=org.id,
+                product=PRDS['rhel'],
+                repo=value['repo'],
+                reposet=value['reposet'],
+                releasever=None,
+            ))
+            # step 3.2: sync repository
+            repository.sync()
+            repos.append(repository)
+        # step 4: Create content view
+        content_view = entities.ContentView(organization=org).create()
+        cls.cv_name = content_view.name
+        # step 5: Associate repository to new content view
+        content_view.repository = repos
+        content_view = content_view.update(['repository'])
+        # step 6.1: Publish content view
+        content_view.publish()
+        # step 6.2: Promote content view to lifecycle_env
+        content_view = content_view.read()
+        # self.assertEqual(len(content_view.version), 1)
+        promote(content_view.version[0], env.id)
+        # step 7: Create activation key
+        activation_key = entities.ActivationKey(
+            name=cls.ak_name,
+            environment=env,
+            organization=org,
+            content_view=content_view,
+        ).create()
+        # step 7.1: Walk through the list of subscriptions.
+        # Find the "Red Hat Employee Subscription" and attach it to the
+        # recently-created activation key.
+        for sub in entities.Subscription(organization=org).search():
+            if sub.read_json()['product_name'] == DEFAULT_SUBSCRIPTION_NAME:
+                # 'quantity' must be 1, not subscription['quantity']. Greater
+                # values produce this error: "RuntimeError: Error: Only pools
+                # with multi-entitlement product subscriptions can be added to
+                # the activation key with a quantity greater than one."
+                activation_key.add_subscriptions(data={
+                    'quantity': 1,
```

Comment:
I think we may need it, just ported this code from api-smoke. This test takes almost an hour.  I think no harm anyways.